ecdsa/src/curve/secp256k1/recoverable_signature.rs

```diff
@@ -0,0 +1,101 @@
+//! Ethereum-style "recoverable signatures"
+
+use super::FixedSignature;
+use crate::{
+    generic_array::{typenum::U64, GenericArray},
+    Error,
+};
+use core::{
+    convert::{TryFrom, TryInto},
+    fmt::{self, Debug},
+};
+
+/// Size of an Ethereum-style recoverable signature in bytes
+pub const SIZE: usize = 65;
+
+/// Ethereum-style "recoverable signatures" which allow for the recovery of
+/// the signer's [`PublicKey`] from the signature itself.
+///
+/// These consist of [`FixedSignature`] followed by a 1-byte [`RecoveryId`]
+/// (65-bytes total)
```

Comment:
It'd be nice to reference some kind of specification here. Unfortunately I wasn't able to find one. Anybody know of one?

Reply:
The only signature specification I was able to find is [EIP-2098](https://eips.ethereum.org/EIPS/eip-2098) which describes a new 64-byte signature format which is able to take advantage of low-s normalization to pack the recovery ID into the `s` value.

It seems there isn't a definitive spec on this format anywhere otherwise. However I've double checked the message structure against a few places where it's documented, like this (see "Extracting the Signature Parameters" section):

https://programtheblockchain.com/posts/2018/02/17/signing-and-verifying-messages-in-ethereum/